File: AverageVectorValue/source.py
import datetime

from absl import app, flags
from absl.flags import FLAGS

flags.DEFINE_string('filename', 'file_large.txt', 'filename to save the vector to')


def read_vector_from_file(filename: str):
    with open(filename) as f:
        n = int(f.readline())
        numbers = [int(number) for number in f.readline().split()]
    return n, numbers


def main(_argv):
    start_time = datetime.datetime.now()

    n, numbers = read_vector_from_file(FLAGS.filename)
    input_output_time = datetime.datetime.now()
    input_output_duration = (input_output_time - start_time).total_seconds() * (10**6)  # microseconds
    print("Reading the vector from file took {} microseconds".format(input_output_duration))

    list_sum = 0
    for number in numbers:
        list_sum += number
    print("Total sum is {}".format(list_sum))
    average_value = (1.0 * list_sum) / n
    print("The average value of the vector is {}".format(average_value))

    final_time = datetime.datetime.now()
    computation_duration = (final_time - input_output_time).total_seconds() * (10**6)
    total_duration = (final_time - start_time).total_seconds() * (10**6)

    print("Processing (computing time) took {} microseconds".format(computation_duration))
    print("Total program duration was {} microseconds".format(total_duration))


if __name__ == "__main__":
    try:
        app.run(main)
    except SystemExit:
        pass

File: AverageVectorValue/source.jl


function read_vector_from_file(filename)
    open(filename) do io
        n = parse(Int, readline(io))
        numbers = [parse(Int, number) for number in split(readline(io))]
        return n, numbers
    end

end

function main()
    starting_unix_time = time()

    filename = "file_large.txt"
    if !isempty(ARGS)
        filename = ARGS[1]
    end
    n, numbers = read_vector_from_file(filename)

    input_output_time = time()
    input_output_duration = trunc(Int, (input_output_time - starting_unix_time) * (10^6))
    println("Reading the vector from file took $input_output_duration  microseconds")

    list_sum = 0
    for number in numbers
        list_sum += number
    end
    println("Total sum is $list_sum")

    average_value = (1.0 * list_sum) / n
    println("The average value of the vector is $average_value")
    final_time = time()
    computation_duration = trunc(Int, (final_time - input_output_time) * (10^6))
    total_duration = trunc(Int, (final_time - starting_unix_time) * (10^6))

    println("Processing (computing time) took $computation_duration microseconds")
    println("Total program duration was $total_duration microseconds")

end


main()



